src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -247,11 +247,6 @@ public void Dispose()
 
                 var buffer = result.Buffer;
 
-                if (result.IsCanceled)
-                {
-                    break;
-                }
```

Comment:
This breaks our HTTP/1.1 abort semantics. We intend to cut of data in this case. Don't cancel the pipe reader in HTTP/2 if you don't want to abruptly cut of the transmission.